mapreader/download/iiif_downloader.py

```diff
@@ -0,0 +1,396 @@
+        """
+        if not isinstance(iiif, list):
+            iiif = [iiif]
+        if iiif_versions is None:
+            iiif_versions = [None] * len(iiif)
+        if not isinstance(iiif_versions, list):
+            iiif_versions = [iiif_versions]
+        if len(iiif) != len(iiif_versions):
+            raise ValueError(
+                "[ERROR] Length of `iiif` must match length of `iiif_versions`."
+            )
+
+        iiif_objs = []
+        for iiif_obj, iiif_version in zip(iiif, iiif_versions):
+            if isinstance(iiif_obj, str):
+                iiif_obj = load_iiif_presentation(iiif_obj, iiif_version)
+            if isinstance(iiif_obj, (IIIFPresentation3, IIIFPresentation2)):
+                iiif_objs.append(iiif_obj)
+            else:
+                raise ValueError("[ERROR] Each `iiif` must be a string or IIIF object.")
+
+        if iiif_uris is None:
+            iiif_uris = [None] * len(iiif_objs)
+        if not isinstance(iiif_uris, list):
+            iiif_uris = [iiif_uris]
+        if len(iiif_uris) != len(iiif_objs):
+            raise ValueError(
+                "[ERROR] Length of `iiif_uris` must match length of `iiif`."
+            )
+
+        for iiif_obj, iiif_uri in zip(iiif_objs, iiif_uris):
+            if not hasattr(iiif_obj, "id"):
+                if iiif_uri is not None:
+                    iiif_obj.id = iiif_uri
+                else:
+                    raise ValueError(
+                        "[ERROR] One of your IIIF objects is missing an 'id' field so we cannot identify it's URL. Please pass the `iiif_uris` argument."
+                    )
+
+        self.iiif = iiif_objs
+
+    def save_georeferenced_maps(
+        self,
+        path_save: str | pathlib.Path = "maps",
+    ):
+        """Save georefereced maps from a IIIF presentation 3 json as tiffs.
+
+        Parameters
+        ----------
+        path_save : str | pathlib.Path
+            Path to save the images
+
+        Notes
+        -----
+        Calls the `save_georeferenced_map` method for each IIIF object.
+        """
+        for iiif in self.iiif:
+            self.save_georeferenced_map(iiif, path_save=path_save)
+
+    def save_georeferenced_map(
+        self,
+        iiif: str | IIIFPresentation3,
+        path_save: str | pathlib.Path = "maps",
+        iiif_uri: str = None,
+    ):
+        """Save a single georefereced map from a IIIF presentation 3 json as a tiff.
+
+        Parameters
+        ----------
+        iiif : str | dict | IIIFPresentation3
+            the IIIF url or IIIF presentation API object containing the map(s)
+        path_save : str | pathlib.Path
+            Path to save the images
+        iiif_uri : str
+            The URI of the IIIF object, needed if the IIIF object is missing an 'id' field.
+
+        Raises
+        ------
+        ValueError
+            Only first order polynomial transformations are currently supported.
```

Comment:
Maybe include link to allmaps docs about this and perhaps a todo note here about supporting other transformations in the future? (or perhaps just a ticket about this with a pointer to this location in the code and anywhere else relevant)